types/react/canary.d.ts

```diff
@@ -67,7 +67,7 @@ declare module "." {
     export function unstable_useCacheRefresh(): () => void;
 
     interface DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_FORM_ACTIONS {
-        functions: (formData: FormData) => void;
+        functions: (formData: FormData) => void | Promise<void>;
```

Comment:
i think you want this to be Promise<void> | undefined instead

Reply:
why do you think that?